addons/misc/functions/fnc_suitcaseNukeZeus.sqf

```diff
@@ -0,0 +1,211 @@
+                        params ["_device"];
+                        deleteVehicle (_device getVariable [QGVAR(suitcaseNuke_smoke), objNull]);
+                    }];
+                };
+                default {};
+            };
+        };
+
+        // private _customCode = _device getVariable [QGVAR(suitcaseNuke_code), ""];
+        // if(_customCode isNotEqualTo "") then {
+        //     private _codeTarget = _unit getVariable [QGVAR(suitcaseNuke_codeTarget), 0];
+
+        //     if((_codeTarget == 0 and hasInterface) || {
+        //         (_codeTarget == 1 and isServer) || {
+        //         (_codeTarget == 2)}}
+        //     ) then {
+        //         [_device] call (compile _customCode);
+        //     };
+        // };
+    }];
+
+
+    // Defusal Code
+    if(_defuser > 0) then {
+
+        // TODO: with ace, also (optionally) require DefusalKit item
+        private _ace = (isClass(configFile >> "CfgPatches" >> "ace_main"));
+        private _condition = switch(true) do {
+            case (_defuser == 1 and _ace) : { " and ([player] call ace_common_fnc_isEOD)" }; // EOD
+            case (_defuser == 1 and !_ace) : { " and (player getUnitTrait ""ExplosiveSpecialist"")" }; // EOD
+            case (_defuser == 2 and _ace) : { " and (([player] call ace_common_fnc_isEngineer) or ([player] call ace_common_fnc_isEOD)" }; // Engineer
+            case (_defuser == 2 and !_ace) : { " and ((player getUnitTrait ""Engineer"") or (player getUnitTrait ""ExplosiveSpecialist""))" }; // Engineer
+            default { "" }; // Anyone
+        };
+
+
+        [
+            _device,
+            "Defuse Device",
+            "\a3\ui_f_oldman\data\IGUI\Cfg\holdactions\repair_ca.paa",
+            "\a3\ui_f_oldman\data\IGUI\Cfg\holdactions\repair_ca.paa",
+            format ["(_this distance _target < 3) and (alive _target) and _target getVariable [""crowsza_misc_suitcaseNuke_isArmed"", false]%1", _condition],
+            "_caller distance _target < 3 and (alive _target) and _target getVariable [""crowsza_misc_suitcaseNuke_isArmed"", false]",
+            {},
+            {},
+            {
+                _target setVariable [QGVAR(suitcaseNuke_isArmed), false, true];
+                if (isClass(configFile >> "CfgPatches" >> "crowsEW_sounds")) then {
+                    [getPos _target, 200, "device_disarmed", 2] call crowsEW_sounds_fnc_playSoundPos;
+                    [getPos _target, 200, "futuristic_machine_turn_off", 2] call crowsEW_sounds_fnc_playSoundPos;
+                };
+            },
+            {},
+            [],
+            _defuseTime
+        ] remoteExec ["BIS_fnc_holdActionAdd", [0, -2] select isDedicated, _device];
+    };
+};
+
+private _controls = [
+    // TODO: size of slider makes it very difficult to be precise - replace with (parsed) text?
+    ["SLIDER", ["Timer", "How long until the device activates in MM:SS"], [1, 30*60, 5*60, {[_this, "MM:SS"] call BIS_fnc_secondsToString}]],
```

Comment:
would multiplying values with a factor 10 or 100 and then multiply with 0.1 or 0.01 when you get the input, make it better? 
Or set it so minimum step-size would be 10 seconds. So every tick on the slider is 10s. Or 5, etc. Might be easier to use?

Reply:
Minimum step is probably the best approach; the main issue is the wide range of times you might want to set (possibly even above what the max value currently is - I can definitely image scenarios where zeus says "you have one hour to find the device, go!")

Have typed that out, I think I'm inclined to lean towards a text box with a HH:MM:SS format

Reply:
I would say not. Makes it too difficult for Zeus. Imagine forgetting you had 10x acceleration on and putting down a 10s countdown for players. Instantly effect happens. Better to keep it to irl time. (As so far the human players don't experience a 30x acceleration, as the computer does ;-)

Reply:
> On that note - might be useful to have an additional module (or when the current module is reapplied to an existing "device"), that allows the user to pause/restart/adjust the countdown

Ugh, now I'm dealing with race conditions! 😅